tests/components/sfr_box/test_button.py

```diff
@@ -32,17 +30,30 @@ async def test_buttons(
     config_entry_with_auth: ConfigEntry,
     device_registry: dr.DeviceRegistry,
     entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
 ) -> None:
     """Test for SFR Box buttons."""
     await hass.config_entries.async_setup(config_entry_with_auth.entry_id)
     await hass.async_block_till_done()
 
-    check_device_registry(device_registry, EXPECTED_ENTITIES["expected_device"])
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry_with_auth.entry_id
+    )
+    assert device_entries == snapshot
+
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, config_entry_with_auth.entry_id
+    )
+    assert entity_entries == snapshot
+
+    for entity in entity_entries:
+        assert hass.states.get(entity.entity_id) == snapshot(name=entity.entity_id)
 
-    expected_entities = EXPECTED_ENTITIES[Platform.BUTTON]
-    assert len(entity_registry.entities) == len(expected_entities)
 
-    check_entities(hass, entity_registry, expected_entities)
+async def test_reboot(hass: HomeAssistant, config_entry_with_auth: ConfigEntry) -> None:
+    """Test for SFR Box reboot button."""
```

Comment:
I have split this test in half - one generic test to ensure all button entities are created, and one test for the reboot button functionnality.
(There is currently only one button available, but more will be added later.)